### PR TITLE
For #25311 - Set tokens used when communicating with Pocket

### PR DESCRIPTION
**Pull request by Mugurell**
Depends on other unmerged PRs, will rebase when appropriately.
Secrets not set on the server, wanted to first confirm the approach and naming.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -389,6 +389,21 @@ android.applicationVariants.all { variant ->
...
+// -------------------------------------------------------------------------------------------------
+
+    print("Pocket consumer key: ")
+
+    try {
+        def token = new File("${rootDir}/.pocket_consumer_key").text.trim()
+        buildConfigField 'String', 'POCKET_CONSUMER_KEY', '"' + token + '"'
+        println "(Added from .pocket_consumer_key file)"
+    } catch (FileNotFoundException ignored) {
+        buildConfigField 'String', 'POCKET_CONSUMER_KEY', '""'
```
Comment (Amejia481) [inline app/build.gradle:404]: As we don't have secrets for debug builds, should we add debug token here?

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt [hunk 2]
```diff
@@ -1263,6 +1264,15 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = { homescreenSections[HomeScreenSection.POCKET] == true },
     )
 
+    /**
+     * Get the profile id to use in the sponsored stories communications with the Pocket endpoing.
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/utils/Settings.kt:1268]: nit: "endpoint"

**Review by Amejia481 — APPROVED** [review]
LGTM!
Let's coordinate when to add the secrets on the server as they should be add before we land or we could cause build failures.

**Comment by Mugurell** [comment]
Arturo helped with adding the needed token to `secrets`.
Tested that local builds work okay without it.
Seems like this is ready to be merged.

**Pull request merged**